src/SonarQube.Client/Services/SonarQubeService.cs

```diff
@@ -186,93 +107,99 @@ public Uri GetProjectDashboardUrl(string projectKey)
 
             const string ProjectDashboardRelativeUrl = "dashboard/index/{0}";
 
-            return new Uri(this.connection.ServerUri, string.Format(ProjectDashboardRelativeUrl, projectKey));
+            return new Uri(httpClient.BaseAddress, string.Format(ProjectDashboardRelativeUrl, projectKey));
         }
 
-        public async Task<SonarQubeQualityProfile> GetQualityProfileAsync(string projectKey, string organizationKey,
-            SonarQubeLanguage language, CancellationToken token)
-        {
-            EnsureIsConnected();
+        public async Task<IList<SonarQubeOrganization>> GetAllOrganizationsAsync(CancellationToken token) =>
+            await InvokeRequestAsync<IGetOrganizationsRequest, SonarQubeOrganization[]>(token);
 
-            var qualityProfileRequest = new QualityProfileRequest { ProjectKey = projectKey, OrganizationKey = organizationKey };
-            var qualityProfilesResult = await this.sonarqubeClient.GetQualityProfilesAsync(qualityProfileRequest, token);
+        public async Task<IList<SonarQubePlugin>> GetAllPluginsAsync(CancellationToken token) =>
+            await InvokeRequestAsync<IGetPluginsRequest, SonarQubePlugin[]>(token);
 
-            // Special handling for the case when a project was not analyzed yet, in which case a 404 is returned
-            if (qualityProfilesResult.StatusCode == HttpStatusCode.NotFound)
-            {
-                qualityProfileRequest = new QualityProfileRequest { ProjectKey = null, OrganizationKey = organizationKey };
-                qualityProfilesResult = await this.sonarqubeClient.GetQualityProfilesAsync(qualityProfileRequest, token);
-            }
+        public async Task<IList<SonarQubeProject>> GetAllProjectsAsync(string organizationKey, CancellationToken token) =>
+            await InvokeRequestAsync<IGetProjectsRequest, SonarQubeProject[]>(
+                request =>
+                {
+                    request.OrganizationKey = organizationKey;
+                },
+                token);
 
-            qualityProfilesResult.EnsureSuccess();
+        public async Task<IList<SonarQubeProperty>> GetAllPropertiesAsync(CancellationToken token) =>
+            await InvokeRequestAsync<IGetPropertiesRequest, SonarQubeProperty[]>(token);
 
-            var profilesWithGivenLanguage = qualityProfilesResult.Value.Where(x => x.Language == language.Key).ToList();
+        public async Task<SonarQubeQualityProfile> GetQualityProfileAsync(string projectKey, string organizationKey, SonarQubeLanguage language, CancellationToken token)
+        {
+            var qualityProfiles = await InvokeRequestAsync<IGetQualityProfilesRequest, SonarQubeQualityProfile[]>(
+                request =>
+                {
+                    request.ProjectKey = projectKey;
+                    request.OrganizationKey = organizationKey;
+                },
+                token);
+
+            // TODO: add language to the request configuration above
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this 'TODO' comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS1135)